### PR TITLE
chore(platforms): Move the builder to centos 7

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes https://github.com/timberio/vector/issues/2743.

Just to explain this one a bit more...

Glibc is one of the few libs we can't really statically link too. glibc is only forwards compatible, so if we build with glibc 2.0.0, anyone with glibc 1.9.9 won't be able to run Vector. Anyone with a newer version will though!

Previously we used musl, which would build an entirely static dependency.

```
❯ curl -O https://packages.timber.io/vector/0.9.2/vector-x86_64-unknown-linux-musl.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.3M  100 13.3M    0     0  7135k      0  0:00:01  0:00:01 --:--:-- 7132k

vector on  centos-7-builder [$?] is 📦 v0.10.0 via ⬢ v12.18.0 via 🦀 v1.43.1 via ❄️  impure 
❯ tar xvf vector-x86_64-unknown-linux-musl.tar.gz                                        
...
./vector-x86_64-unknown-linux-musl/bin/vector
...
vector on  centos-7-builder [$?] is 📦 v0.10.0 via ⬢ v12.18.0 via 🦀 v1.43.1 via ❄️  impure 
❯ ldd vector-x86_64-unknown-linux-musl/bin/vector                                        
        not a dynamic executable
```

Here's my builds off this new branch:

```
vector on  centos-7-builder [$?] is 📦 v0.10.0 via ⬢ v12.18.0 via 🦀 v1.43.1 via ❄️  impure 
❯ docker run --rm -ti -v $(pwd):/foo centos:7
[root@f37345ab09a7 foo]# rpm -i vector-x86_64.rpm
[root@f37345ab09a7 foo]# ldd /usr/bin/vector 
        linux-vdso.so.1 =>  (0x00007ffe46151000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f6cec17d000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f6cebe76000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f6cebc72000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f6ceba6a000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f6ceb854000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f6ceb486000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6cf0000000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f6ceb184000)
```

Checking the nightlies:

```
[root@f37345ab09a7 /]# curl -O https://packages.timber.io/vector/nightly/latest/vector-x86_64.rpm
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20.0M  100 20.0M    0     0  10.2M      0  0:00:01  0:00:01 --:--:-- 10.2M
[root@f37345ab09a7 /]# .^C
[root@f37345ab09a7 /]# https://packages.timber.io/vector/nightly/latest/vector-x86_64.rpm^C
[root@f37345ab09a7 /]# ldd /usr/bin/vector ^C
[root@f37345ab09a7 /]# rpm -i vector-x86_64.rpm 
error: Failed dependencies:
        libc.so.6(GLIBC_2.18)(64bit) is needed by vector-0.10.0-1.x86_64
        libstdc++.so.6(CXXABI_1.3.8)(64bit) is needed by vector-0.10.0-1.x86_64
```

### CODE REVIEW

**Comment by Hoverbear** [comment]
@MOZGIII

**Comment by MOZGIII** [comment]
I'll need to a thorough look into this. Pretty sure I've had issues with `glibc` recently (about a day ago), but I applied a workaround and moved on.

File: Makefile [hunk 1]
```diff
@@ -413,7 +413,7 @@ package-deb-aarch64: package-archive-aarch64-unknown-linux-musl  ## Build the aa
 # rpms
 
 package-rpm: ## Build the rpm package
-	$(RUN) package-rpm
+	./scripts/package-rpm.sh
```
Comment (MOZGIII) [inline Makefile:416]: `$(RUN) ` prepared target dirs, without it we'll get errors

Comment (Hoverbear) [reply Makefile:416]: Oh dang you're right, let me re-add that!

Comment (MOZGIII) [reply Makefile]: Is this `package-rpm` target used anywhere? I'm pretty sure we're not using it, just like `package-deb`. The purpose of those targets is a mystery to me, so I just don't know what rules we need to follow when maintaining them.
Seems like they're for manual builds when the host OS matches the target env, this we can probably get rid of the `$(RUN)` just as you had it.
So yeah...

Comment (Hoverbear) [reply Makefile]: Yeah it's abit odd. :laughing:

**Comment by Hoverbear** [comment]
Fixed an issue on windows.

**Pull request merged**